dashboard/test/ui/features/learning_platform/script_overview.feature

```diff
@@ -20,11 +19,9 @@ Feature: Script overview page
     And I complete the level on "http://studio.code.org/s/allthethings/lessons/29/levels/4?level_name=2-3 Artist 1 new"
     And I am on "http://studio.code.org/s/allthethings"
     And I wait until element ".teacher-panel" is visible
-    And I wait until jQuery Ajax requests are finished
-    Then I verify progress for lesson 29 level 4 is "perfect"
+    Then I verify progress for lesson 29 level 4 in detail view is "perfect"
```

Comment:
Before adding "in detail view", we were actually verifying the state of a hidden progress bubble.

Reply:
Would it be possible to update verify_progress so that it only queries for visible elements in the DOM?

Reply:
Yes!  In progress.rb, lines 37, 53, and 58 now check for "is visible" in the various cases.